src/content/blog/ddev-name-resolution-wildcards.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Hostnames and Wildcards and DDEV, Oh My!"
+pubDate: 2024-03-30
+#modifiedDate: 2024-04-03
+summary: How hostname name resolution works in a DDEV environment, including how wildcards work and how DNS is involved
+author: Randy Fay
+featureImage:
+  src: /img/blog/2024/03/ddev-name-resolution.png
+  alt: Hostname resolution with DDEV
+  credit: 'Ideogram.ai: Create An illustration showing a simplified and stylized representation of the internet with various devices (computers, tablets, and smartphones) displaying local development websites, all connected by dotted lines to a central server labeled "DDEV". The server has a large, glowing globe above it, symbolizing global DNS, with several dotted lines branching off to represent the resolution of wildcard hostnames. The background is digital and abstract, suggesting connectivity and network infrastructure.'
+categories:
+  - Guides
+---
+
+A very cool thing about DDEV is that you don’t need to think about IP addresses or HTTPS or name resolution most of the time. Some of you will never need to. And you don't have to understand teh rest of this article to use DDEV.
+
+**Most of you will never need to understand anything in this article.**
+
+This article attempts to unwind what DDEV does for you in name resolution, and what to do when things require a bit more effort.
+
+## Name resolution for browsers and URLs
+
+<!-- textlint-disable -->
+The first thing to understand is the structure of a URL used by your browser. A URL like `https://www.google.com/`/search?q=nothing` is structured with 3 main parts, the protocol (usually `https`), the hostname (like `www.google.com`), and the URI, (like `/search?q=nothing`). Every URL that your browser can visit has at least the protocol and the hostname. Sometimes the protocol is hidden implying the default `https`. The URI may be the default "front page".
+<!-- textlint-enable -->
+
+When your browser gets a URL, no matter how it gets it, and you want to display the page at that URL, the browser has to parse the URL, then turn the hostname into an IP address so it can find its way to it on the internet, then do the proper HTTP request to the internet site, including the URI. The key thing for us right now is that it has to look up that hostname.
```

Comment:
```suggestion
When you visit a website, the browser has to: parse the URL, turn the hostname into an IP address (so it can find it on the internet), and do a proper HTTP request including the URI. For now, the key takeaway is that it has to look up the IP address for that hostname.
```